lib/rubocop/rspec/language/runtime_macros.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module RSpec
+    module Language
+      # RuntimeMacros `.def_node_matcher` and `.def_node_search` accept block
+      # that will be evaluated within cop context to get pattern pattern.
+      # It can be used to pattern match values from config because are available
+      # only during runtime.
+      module RuntimeMacros
+        def def_runtime_node_matcher(method_name, &pattern_block)
+          define_method method_name do |node, *args, &block|
```

Comment:
Yeah, even no one in their right mind is going to configure RSpec aliases per directory - I think we should probably keep such behaviour because it is expected behaviour.
I agree, memoizing becomes untrivial, maybe we could use `RSpec/Language` config hash as cache marker. Let's probably wait for some performance impact report?
Also I consider elimination of `SelectorSet` in favour of just arrays of symbols because they are instantiated a lot in runtime matchers implementation. But I'm not sure if it's a performance hit though.

Reply:
> no one in their right mind is going to configure RSpec aliases per directory 

Certain specs, e.g. Pundit/CanCanCan/ActionPolicy specs will reside in their own directory.
Even though it's RSpec aliases are defined globally, this dynamic language resolution is not only about core RSpec DSL aliases.
I believe this feature will also allow to fine-tune detection of custom matchers, negated matchers that are used mostly local to certain directories.